CHANGELOG.md

```diff
@@ -14,6 +14,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Added support for external pids and encoded pids in external terms
 - Added support for external refs and encoded refs in external terms
 - Introduce ports to represent native processes and added support for external ports and encoded ports in external terms
+- Added missing lock on socket structure
```

Comment:
Doesn't this change count as a fix? If so, shouldn't be applied to release-0.6?

Reply:
Right, I'm concerned about this run:
https://github.com/atomvm/AtomVM/actions/runs/12507186602/job/34897717598?pr=1421